### PR TITLE
Port item modifications to 1.16.4

### PR DESCRIPTION
**Pull request by Max094Reikeb**
This procedure ports item modifications to 1.16.4.
- Dispense behavior from #19
- New blockly tooltip support from #93

### CODE REVIEW

File: plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl [hunk 1]
```diff
@@ -60,6 +60,32 @@ package ${package}.item;
...
+				World world = blockSource.getWorld();
+				Direction direction = blockSource.getBlockState().get(DispenserBlock.FACING);
+				int x = blockSource.getBlockPos().getX();
+				int y = blockSource.getBlockPos().getY();
+				int z = blockSource.getBlockPos().getZ();
+
+				this.successful = <@procedureOBJToConditionCode data.dispenseSuccessCondition/>;
+
+				<#if hasProcedure(data.dispenseResultItemstack)>
+					boolean success = this.successful;
```
Comment (NOYBss) [inline plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl:79]:
```suggestion
				this.setSuccessful(<@procedureOBJToConditionCode data.dispenseSuccessCondition/>);

				<#if hasProcedure(data.dispenseResultItemstack)>
					boolean success = this.isSuccessful();

```

**Review by NOYBss — CHANGES_REQUESTED** [review]
This should fix the crash. Looks like mojang added methods for setting/getting.

File: plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl (hunk 1 above)
Comment (Max094Reikeb) [reply plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl]: Done, but I think there is another error...

File: plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl [hunk 2]
```diff
@@ -60,6 +60,32 @@ package ${package}.item;
...
+			public ItemStack dispenseStack(IBlockSource blockSource, ItemStack stack) {
+				this.successful = true;
+				ItemStack itemstack = stack.copy();
+				World world = blockSource.getWorld();
+				Direction direction = blockSource.getBlockState().get(DispenserBlock.FACING);
+				int x = blockSource.getBlockPos().getX();
+				int y = blockSource.getBlockPos().getY();
+				int z = blockSource.getBlockPos().getZ();
+
+				this.successful = <@procedureOBJToConditionCode data.dispenseSuccessCondition/>;
```
Comment (NOYBss) [inline plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl:76]:
```suggestion
				this.setSuccessful(<@procedureOBJToConditionCode data.dispenseSuccessCondition/>);
```
You forgot this part 😄

**Review by NOYBss — CHANGES_REQUESTED** [review]

File: plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl (hunk 2 above)
Comment (Max094Reikeb) [reply plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl]: Done!

File: plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl [hunk 3]
```diff
@@ -60,6 +60,32 @@ package ${package}.item;
 		elements.items.add(() -> new ItemCustom());
 	}
 
+	<#if data.hasDispenseBehavior>
+	@Override
+	public void init(FMLCommonSetupEvent event) {
+		DispenserBlock.registerDispenseBehavior(this.block, new OptionalDispenseBehavior() {
+			public ItemStack dispenseStack(IBlockSource blockSource, ItemStack stack) {
+				this.successful = true;
```
Comment (NOYBss) [inline plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl:68]:
```suggestion
				this.setSuccessful(true);
```
I didn't see this either 😅

**Review by NOYBss — CHANGES_REQUESTED** [review]
I didn't see this either 😅

File: plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl (hunk 3 above)
Comment (Max094Reikeb) [reply plugins/generator-1.16.4/forge-1.16.4/templates/item.java.ftl]: Done!

**Comment by NOYBss** [comment]
Queues takes forever... Anyways I don't see any other direct access to `successful` boolean

**Comment by AlsoSomeoneElse** [comment]
Seems good, but rather than edit the procedures.java.ftl file here too, I think it'd be better to wait for #131 to get approved instead

**Review by NOYBss — APPROVED** [review]
Looks good now, but 2 pr's adding same stuff will conflict (even though its the same thing). So when the first pr gets merged, second must resolve the conflicts.

**Comment by Max094Reikeb** [comment]
> Looks good now, but 2 pr's adding same stuff will conflict (even though its the same thing). So when the first pr gets merged, second must resolve the conflicts.

Yes I know

**Comment by NOYBss** [comment]
`procedureToRetvalCode` and `procedureToRetvalCodeTwo` are the same thing, there is only name difference, whats two for?

**Comment by Max094Reikeb** [comment]
> `procedureToRetvalCode` and `procedureToRetvalCodeTwo` are the same thing, there is only name difference, whats two for?

Let it go, I'm deleting my changes anyways. The last commit of @AlsoSomeoneElse added everything I needed so I'm gonna revert my changes.

**Pull request merged**